api/v1/verticadb_webhook.go

```diff
@@ -2059,6 +2111,27 @@ func (v *VerticaDB) checkImmutableStsName(oldObj *VerticaDB, allErrs field.Error
 	return allErrs
 }
 
+// checkImmutableClientProxy will validate the proxy spec fields in vdb
+func (v *VerticaDB) checkImmutableClientProxy(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
+	// annotation vertica.com/use-client-proxy cannot change after creation
+	if v.Annotations[vmeta.UseVProxyAnnotation] != oldObj.Annotations[vmeta.UseVProxyAnnotation] {
+		prefix := field.NewPath("metadata").Child("annotations")
+		errMsg := fmt.Sprintf("annotation %s cannot change after creation", vmeta.UseVProxyAnnotation)
+		err := field.Invalid(prefix.Key(vmeta.UseVProxyAnnotation),
+			v.Annotations[vmeta.VClusterOpsAnnotation],
+			errMsg)
+		allErrs = append(allErrs, err)
+	}
+	// proxy.image cannot change after creation
+	if v.Spec.Proxy != nil && oldObj.Spec.Proxy != nil && v.Spec.Proxy.Image != oldObj.Spec.Proxy.Image {
```

Comment:
need to check one more condition: `oldObj.Spec.Proxy != nil && v.Spec.Proxy == nil`

Reply:
If we add this, we will end up not able to remove the proxy field. Say we disabled the proxy by editing the annotation, the old spec has the proxy fields, and the new proxy is empty, it will always fail on this check.

Reply:
You cannot change the annotation after vdb is created.

Reply:
I see, condition added in https://github.com/vertica/vertica-kubernetes/pull/1000/commits/24fc8b2393034aecdab40e3eb8cf877171b88f31